src/script/components/calling/FullscreenVideoCall.tsx

```diff
@@ -125,35 +126,132 @@ const FullscreenVideoCall: React.FC<FullscreenVideoCallProps> = ({
     'classifiedDomains',
   ]);
 
-  const {[MediaDeviceType.VIDEO_INPUT]: currentCameraDevice, [MediaDeviceType.AUDIO_INPUT]: currentMicrophoneDevice} =
-    useKoSubscribableChildren(mediaDevicesHandler.currentDeviceId, [
-      MediaDeviceType.VIDEO_INPUT,
-      MediaDeviceType.AUDIO_INPUT,
-    ]);
-
-  const {videoinput, audioinput} = useKoSubscribableChildren(mediaDevicesHandler.availableDevices, [
+  const {
+    [MediaDeviceType.VIDEO_INPUT]: currentCameraDevice,
+    [MediaDeviceType.AUDIO_INPUT]: currentMicrophoneDevice,
+    [MediaDeviceType.AUDIO_OUTPUT]: currentSpeakerDevice,
+  } = useKoSubscribableChildren(mediaDevicesHandler.currentDeviceId, [
     MediaDeviceType.VIDEO_INPUT,
     MediaDeviceType.AUDIO_INPUT,
+    MediaDeviceType.AUDIO_OUTPUT,
   ]);
 
+  const {videoinput, audioinput, audiooutput} = useKoSubscribableChildren(mediaDevicesHandler.availableDevices, [
+    MediaDeviceType.VIDEO_INPUT,
+    MediaDeviceType.AUDIO_INPUT,
+    MediaDeviceType.AUDIO_OUTPUT,
+  ]);
+  const [audioOptionsOpen, setAudioOptionsOpen] = React.useState(false);
+  const [videoOptionsOpen, setVideoOptionsOpen] = React.useState(false);
   const minimize = () => multitasking.isMinimized(true);
 
   const showToggleVideo =
     isVideoCallingEnabled &&
     (call.initialType === CALL_TYPE.VIDEO || conversation.supportsVideoCall(call.isConference));
   const availableCameras = useMemo(
-    () =>
-      videoinput.map(device => (device as MediaDeviceInfo).deviceId || (device as ElectronDesktopCapturerSource).id),
+    () => videoinput.map(device => (device as MediaDeviceInfo) || (device as ElectronDesktopCapturerSource)),
     [videoinput],
   );
-  const showSwitchCamera = availableCameras.length > 1;
 
   const availableMicrophones = useMemo(
-    () =>
-      audioinput.map(device => (device as MediaDeviceInfo).deviceId || (device as ElectronDesktopCapturerSource).id),
+    () => audioinput.map(device => (device as MediaDeviceInfo) || (device as ElectronDesktopCapturerSource)),
```

Comment:
question: Do we really need this mapping here? You actually allow the type to be device: `ElectronDesktopCapturerSource | MediaDeviceInfo` when mapping in `audioOptions` below 🤔

Reply:
ive removed this